nomenclature/codelist.py

```diff
@@ -116,7 +117,7 @@ def validate_items(self, items: List[str]) -> List[str]:
         list
             Returns the list of items that are **not** defined in the codelist
         """
-        return [c for c in items if c not in self.keys()]
+        return [item for item in items if not any(fnmatch(item, code) for code in self)]
```

Comment:
**fnmatch** does not work work square brackets as discussed in https://github.com/IAMconsortium/pyam/issues/542, so I suggest to use **pyam.utils.pattern_match** for consistent behaviour with pyam-filtering.